src/layout/Table/typeguards.ts

```diff
@@ -0,0 +1,16 @@
+import type { IDataModelReference } from 'src/layout/common.generated';
+
+export function isIDataModelReference(obj: unknown): obj is IDataModelReference {
+  return (
+    typeof obj === 'object' &&
+    obj !== null &&
+    'dataType' in obj &&
+    'field' in obj &&
+    typeof (obj as { dataType: unknown }).dataType === 'string' &&
+    typeof (obj as { field: unknown }).field === 'string'
+  );
+}
+
+export function isIDataModelReferenceArray(obj: unknown): obj is IDataModelReference[] {
+  return Array.isArray(obj) && obj.every(isIDataModelReference);
+}
```

Comment:
What's this for? Arrays of data model references would be an entirely new concept unknown to me at least.